src/react/tests/useQueries.test.tsx

```diff
@@ -35,11 +43,129 @@ describe('useQueries', () => {
 
     renderWithClient(queryClient, <Page />)
 
-    await sleep(30)
+    await sleep(10)
 
     expect(results.length).toBe(3)
     expect(results[0]).toMatchObject([{ data: undefined }, { data: undefined }])
     expect(results[1]).toMatchObject([{ data: 1 }, { data: undefined }])
     expect(results[2]).toMatchObject([{ data: 1 }, { data: 2 }])
   })
+
+  it('should return the correct states with suspense', async () => {
+    const key1 = queryKey()
+    const key2 = queryKey()
+    const results: UseQueryResult[][] = []
+
+    function Page() {
+      const result = useQueries([
+        {
+          queryKey: key1,
+          queryFn: async () => {
+            await sleep(5)
+            return 1
+          },
+          suspense: true,
+        },
+        {
+          queryKey: key2,
+          queryFn: async () => {
+            await sleep(7)
+            return 2
+          },
+        },
+      ])
+      results.push(result)
+      return null
+    }
+
+    renderWithClient(
+      queryClient,
+      <React.Suspense fallback={'test'}>
+        <Page />
+      </React.Suspense>
+    )
+
+    await sleep(10)
+
+    expect(results.length).toBe(1)
+    expect(results[0]).toMatchObject([{ data: 1 }, { data: 2 }])
```

Comment:
i'd expect more from this test. If you are playing with `suspense: true/false(undefined)` queries why not check intermediate values like `[{ data: 1}, { loading: true }]` in this case?

Reply:
And actually suspending on first query :)

Reply:
Thank you for the review!

What I think is happening here is that it _is_ suspending, so fallback is rendering until the hook is no longer throwing, at which point it does its first render with actual data.  Are there intermediate states I would expect to see on first render?  Would a refetch to show new states after original suspense render help to make this case more robust?

Reply:
What i mean is: the component suspends on first render, but 5s later it render again and push `[{ data: 1}, { loading: true }]` in results. This should happen because first query suspends since it has `suspense: true` but the second query behaves like regular `useQuery(...{ suspense: false })` doesn't it?

Reply:
Ah, I see what you mean.  The initial PR suspended when any of the queries was configured for suspense and any of the queries was loading. In other words, if anything had suspense, everything had suspense.  

This could be revised so that it only suspends when any of the queries that are configured for suspense are loading--is that the expected functionality? I'm honestly not sure what a client would expect as behavior here. I feel the same about the current summary error handling.

Reply:
Yeah, the test should go something like:
- Suspends on first load
- Data comes in
- query becomes stale
- query gets refetched
- `isFetching === true`

Reply:
Revisiting this, apologies for the delay. I do feel a bit out of context on this one : )

Is the desired behavior that all queries have aggregate suspense and error state _until_ the first settlement of all queries, and then queries behave independently? 

My concern with this approach is that the individual queries still accept a fully query config, including options like suspense, but they aren't guaranteed to honor that config (and won't individually honor certain configs like `suspense`).

I wonder if this is a case where `useQueries` and `useQueriesSuspense` should exist as distinct hooks that can explicitly define their option subsets and be opinionated for suspense vs no suspense.

Reply:
> I wonder if this is a case where useQueries and useQueriesSuspense should exist as distinct hooks that can explicitly define their option subsets and be opinionated for suspense vs no suspense.

Hmm, I wouldn't be a big fan of this api wise. For example, because we just got #2395 reported: `useErrorBoundary` alone also doesn't work with `useQueries`, and I think the reasons are similar. I tried the reproduction out with this PR and it seems to work.

Now we have to think about if we even want different behaviour for multiple queries in the same `useQueries` batch. E.g. one of them having suspense, while others don't. Or one has useErrorBoundary and another one doesn't.

I would say this shouldn't be allowed, kind of like an invalid state. The current api makes this possible because it's just an array of useQuery configs. If the structure were a bit different, it would be more obvious, something like:

```
useQueries({
    queries: [{ queryKey1, queryFn1, options1 }, { queryKey2, queryFn2, options2 }],
    suspense: boolean,
    useErrorBoundary: boolean
})
```

with that, we would lift "global" options out of the array, so you can only set them once for all queries. The options on each query would then _not_ allow what we have at the top level, e.g. suspense or useErrorBoundary.

---

I don't to propose a breaking change to the useQueries api though, just some food for thought. I am personally fine with doing what we do now, which is:

```
const someSuspense = defaultedQueries.some((q) => q.suspense);
const someUseErrorBoundary = defaultedQueries.some((q) => q.useErrorBoundary);
```

But we'd need to document that setting `suspense` or `useErrorBoundary` to true on _any_ of the queries will affect all queries! It's not that intuitive, but we can revisit it for v4 maybe.